src/Markdig/Helpers/StringLineGroup.cs

```diff
@@ -187,9 +187,33 @@ public void Trim()
         }
     }
 
+    private struct Enumerator : IEnumerator
```

Comment:
Could you add a `public Enumerator GetEnumerator()` to avoid the boxing of this struct?

Reply:
Do you mean turn explicit implementation of `IEnumerator.GetEnumerator()` into `Enumerator GetEnumerator()`? I guess it's not intended for directly use since it's explicit impl.

Reply:
No. It's called (enumerator) duck-typing. You keep the explicit implementation but you declare a public method with the same signature but that returns the struct `Enumerator GetEnumerator()`.

When using foreach, it will detect this method and avoid calling the explicit implementation.

Reply:
Added.